core/src/main/java/com/orientechnologies/orient/core/index/OIndexManagerShared.java

```diff
@@ -521,7 +511,7 @@ public void run() {
         }
 
         try {
-          recreateIndexes(storage, indexesToRebuild, newDb);
+          recreateIndexes(indexesToRebuild, newDb);
         } finally {
           if (storage instanceof OAbstractPaginatedStorage) {

```

Comment:
Unsafe mode removed. Index rebuilding left as it is with minor modifications, will create a separate issue.